web/fns.php

```diff
@@ -140,34 +146,68 @@ function alloy_query($type, $design, $attributes, $bbox, $join_attributes=null)
                 "dodiCode" => $design,
                 "attributes" => $attributes,
             ],
-            "children"=> [[
-                "type" => "GeomIntersects",
-                "children" => [
-                    [
-                        "type" => "Attribute",
-                        "properties" => [ "attributeCode" => "attributes_itemsGeometry" ]
-                    ],
-                    [
-                        "type" => "Geometry",
-                        "properties" => [
-                            "value" => [
-                                "type" => "Polygon",
-                                "coordinates" => [[
-                                    [$bbox[0], $bbox[1]],
-                                    [$bbox[0], $bbox[3]],
-                                    [$bbox[2], $bbox[3]],
-                                    [$bbox[2], $bbox[1]],
-                                    [$bbox[0], $bbox[1]],
-                                ]]
-                            ]
+            "children" => []
+        ]
+    ];
+
+    $geometry =  [
+        "children"=> [[
+            "type" => "GeomIntersects",
+            "children" => [
+                [
+                    "type" => "Attribute",
+                    "properties" => [ "attributeCode" => "attributes_itemsGeometry" ]
+                ],
+                [
+                    "type" => "Geometry",
+                    "properties" => [
+                        "value" => [
+                            "type" => "Polygon",
+                            "coordinates" => [[
+                                [$bbox[0], $bbox[1]],
+                                [$bbox[0], $bbox[3]],
+                                [$bbox[2], $bbox[3]],
+                                [$bbox[2], $bbox[1]],
+                                [$bbox[0], $bbox[1]],
+                            ]]
                         ]
                     ]
                 ]
-            ]]
-        ]
+            ]
+        ]]
     ];
+
+    $filter = [
```

Comment:
This could all be in the if below, it's only needed if pre_fetch_filter is set (and then you could put the geometry directly in $filter's children rather than adding it).